src/os/filestore/FileJournal.cc

```diff
@@ -134,6 +138,7 @@ int FileJournal::_open(bool forwrite, bool create)
 
  out_fd:
   VOID_TEMP_FAILURE_RETRY(::close(fd));
+  fd = -1;
```

Comment:
I led you astray.  Actually, the open doesn't fail, so without the S_ISREG check, it would crash later upon read failure.